components/script/dom/location.rs

```diff
@@ -249,8 +352,14 @@ impl LocationMethods for Location {
 
     // https://html.spec.whatwg.org/multipage/#dom-location-search
     fn SetSearch(&self, value: USVString) -> ErrorResult {
-        self.check_same_origin_domain()?;
-        self.set_url_component(value, UrlHelper::SetSearch);
+        // Step 1: If this Location object's relevant Document is null, then return.
+        if self.window.has_document() {
+            // Step 2: If this Location object's relevant Document's origin is not
+            // same origin-domain with the entry settings object's origin, then
+            // throw a "SecurityError" DOMException.
+            self.check_same_origin_domain()?;
+            self.set_url_component(value, UrlHelper::SetSearch);
```

Comment:
I think we're missing step 4 and 5 here?

Reply:
Again, I presumed those steps as a part of `UrlHelper::SetSearch`.